include/deal.II/base/tensor.h

```diff
@@ -761,10 +776,10 @@ std::ostream &operator << (std::ostream &out, const Tensor<rank_,1> &p)
  * @relates Tensor
  * @author Guido Kanschat, 2000
  */
-template <int dim, typename Number>
+template <int dim, typename Number, typename OtherNumber>
 inline
 Number contract (const Tensor<1,dim,Number> &src1,
-                 const Tensor<1,dim,Number> &src2)
+                 const Tensor<1,dim,OtherNumber> &src2)
```

Comment:
The return type of this should now be `typename ProductType<Number,OtherNumber>::type`. Same in the `operator*`.